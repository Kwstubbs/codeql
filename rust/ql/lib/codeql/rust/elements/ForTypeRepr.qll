// generated by codegen, do not edit
/**
 * This module provides the public class `ForTypeRepr`.
 */

private import internal.ForTypeReprImpl
import codeql.rust.elements.ForBinder
import codeql.rust.elements.TypeRepr

/**
 * A type with a higher-ranked `for` modifier. This is currently not valid Rust syntax (`for<...>` can
 * only be applied to traits to form a `TypeBound`).
 *
 * For example:
 * ```rust
 * fn foo(value: for<'a> usize) {}  // DOESN'T COMPILE
 * //            ^^^^^^^^^^^^^
 * ```
 */
final class ForTypeRepr = Impl::ForTypeRepr;
