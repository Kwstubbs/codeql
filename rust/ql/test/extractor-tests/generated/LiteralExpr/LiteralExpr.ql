// generated by codegen, do not edit
import codeql.rust.elements
import TestUtils

query predicate instances(
  LiteralExpr x, string getNumberOfAttrs__label, int getNumberOfAttrs, string hasTextValue__label,
  string hasTextValue
) {
  toBeTested(x) and
  not x.isUnknown() and
  getNumberOfAttrs__label = "getNumberOfAttrs:" and
  getNumberOfAttrs = x.getNumberOfAttrs() and
  hasTextValue__label = "hasTextValue:" and
  if x.hasTextValue() then hasTextValue = "yes" else hasTextValue = "no"
}

query predicate getAttr(LiteralExpr x, int index, Attr getAttr) {
  toBeTested(x) and not x.isUnknown() and getAttr = x.getAttr(index)
}

query predicate getTextValue(LiteralExpr x, string getTextValue) {
  toBeTested(x) and not x.isUnknown() and getTextValue = x.getTextValue()
}
