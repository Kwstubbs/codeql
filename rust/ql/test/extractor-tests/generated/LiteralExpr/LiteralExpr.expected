instances
| gen_literal_expr.rs:5:5:5:6 | 42 | getNumberOfAttrs: | 0 | hasTextValue: | yes |
| gen_literal_expr.rs:6:5:6:8 | 42.0 | getNumberOfAttrs: | 0 | hasTextValue: | yes |
| gen_literal_expr.rs:7:5:7:19 | "Hello, world!" | getNumberOfAttrs: | 0 | hasTextValue: | yes |
| gen_literal_expr.rs:8:5:8:20 | b"Hello, world!" | getNumberOfAttrs: | 0 | hasTextValue: | yes |
| gen_literal_expr.rs:9:5:9:7 | 'x' | getNumberOfAttrs: | 0 | hasTextValue: | yes |
| gen_literal_expr.rs:10:5:10:8 | b'x' | getNumberOfAttrs: | 0 | hasTextValue: | yes |
| gen_literal_expr.rs:11:5:11:20 | r"Hello, world!" | getNumberOfAttrs: | 0 | hasTextValue: | yes |
| gen_literal_expr.rs:12:5:12:8 | true | getNumberOfAttrs: | 0 | hasTextValue: | yes |
getAttr
getTextValue
| gen_literal_expr.rs:5:5:5:6 | 42 | 42 |
| gen_literal_expr.rs:6:5:6:8 | 42.0 | 42.0 |
| gen_literal_expr.rs:7:5:7:19 | "Hello, world!" | "Hello, world!" |
| gen_literal_expr.rs:8:5:8:20 | b"Hello, world!" | b"Hello, world!" |
| gen_literal_expr.rs:9:5:9:7 | 'x' | 'x' |
| gen_literal_expr.rs:10:5:10:8 | b'x' | b'x' |
| gen_literal_expr.rs:11:5:11:20 | r"Hello, world!" | r"Hello, world!" |
| gen_literal_expr.rs:12:5:12:8 | true | true |
